Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -196,8 +196,22 @@ public void Delete(string id)
                                 "The current deployment branch is '{0}', but nothing has been pushed to it",
```

Comment:
@sanchitmehta : Question for you (unrelated to this change) - Should we start using `RestartMainSiteAsync` instead of `RequestContainerRestart` unconditionally in `RestartMainSiteIfNeeded`? I remember you mentioning there are occasional issues with requesting restart using file system based restart triggers.

Reply:
@shrishrirang I had a discussion about this, main problem moving with that is a 45 seconds delay in getting the change notification in LWAS - any increase in the frequency can impact DataRole. I'm definitely still on-board with it, but want to load test it a bit first. So yes for future we can consider this.

Reply:
Sounds good. Thanks!